### PR TITLE
Proposed fix for Issue #836 - height and width parameters ignored when barcode_format is DATA_MATRIX

### PR DESCRIPTION
**Pull request by kent92**
Hi @srowen , was wondering if you'd like to go over my proposed fix for this issue. Tried it locally and it works, passes unit tests, etc. Let me know if further changes are needed and i will follow through accordingly. 

Thanks!

### CODE REVIEW

**Comment by axxel** [comment]
My 2 cents as an outsider: why not refactor the QRCode + AZtech + DataMatrix writers so they all use the same simple scaling/inflating code instead of adding yet another copy of the functionality? (I'm biased, of course: https://github.com/zxing/zxing/issues/836#issuecomment-345980338 ;))

File: core/src/main/java/com/google/zxing/datamatrix/DataMatrixWriter.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
  */
 public final class DataMatrixWriter implements Writer {
 
+
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/datamatrix/DataMatrixWriter.java]: If you would, revert the extra space changes like this one and one below

File: core/src/main/java/com/google/zxing/datamatrix/DataMatrixWriter.java [hunk 2]
```diff
@@ -159,17 +160,33 @@ private static BitMatrix encodeLowLevel(DefaultPlacement placement, SymbolInfo s
    * @param matrix The input matrix.
    * @return The output matrix.
    */
-  private static BitMatrix convertByteMatrixToBitMatrix(ByteMatrix matrix) {
-    int matrixWidgth = matrix.getWidth();
+  private static BitMatrix convertByteMatrixToBitMatrix(ByteMatrix matrix, int reqWidth, int reqHeight) {
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/datamatrix/DataMatrixWriter.java]: It's worth a little javadoc to clarify what these two args are, especially as they're abbreviated

File: core/src/main/java/com/google/zxing/datamatrix/DataMatrixWriter.java [hunk 3]
```diff
@@ -159,17 +160,33 @@ private static BitMatrix encodeLowLevel(DefaultPlacement placement, SymbolInfo s
...
+    int multiple = Math.min(outputWidth / matrixWidth, outputHeight / matrixHeight);
+
+    int leftPadding = (outputWidth - (matrixWidth * multiple)) / 2 ;
+    int topPadding = (outputHeight - (matrixHeight * multiple)) / 2 ;
+
+    BitMatrix output;
+
+    // remove padding if requested width and height are too small
+    if (reqHeight < matrixHeight || reqWidth < matrixWidth) {
+      leftPadding = topPadding = 0 ;
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/datamatrix/DataMatrixWriter.java]: separate these into two statements and remove the space before ;

**Comment by kent92** [comment]
@srowen added the changes you've suggested. Let me know if there's more. Thanks

**Pull request merged**